### DISCUSSION TITLE
📣 Releasing r33

### DISCUSSION THREAD

**Discussion by joelim-work** [gokcehan/lf#1817] [General]
It has been over half a year since [r32](https://github.com/gokcehan/lf/releases/tag/r32) was released, and I think it is a good time now to release a new version with all the changes that have happened since then. It appears that @gokcehan hasn't had time recently to oversee the new changes, however as a collaborator I wish to keep the project going so that the wider community can benefit from them.

This is the first time that I am creating such a release, but I believe the steps should be straightforward from the [defined workflow](https://github.com/gokcehan/lf/blob/master/.github/workflows/release.yml):

- Run [`doc-with-docker.sh`](https://github.com/gokcehan/lf/blob/master/gen/doc-with-docker.sh) and ensure that the generated documentation is up to date.
- Create a new lightweight tag and push it, triggering the release workflow.
- Update the release notes.

I am planning to make the new release in around a couple of weeks. and will be staging a change freeze until that point. If anyone has any queries or concerns, **please let me know**.

Below are the release notes I am planning to include:

```md
## Changelog

- **(Breaking)** The `globsearch` option, which previously affected both searching and filtering, now affects only searching. A new `globfilter` option is introduced to enable globs when filtering, and acts independently from `globsearch`. (#1650) (by @Catalyn45)
- **(Breaking)** The `hidecursorinactive` option is replaced by the `on-focus-gained` and `on-focus-lost` hook commands. These commands can be used to invoke custom behavior when the terminal gains or loses focus. (#1763) (by @joelim-work)
- **(Breaking)** The `ruler` option (deprecated in favor of `rulerfmt`) is now removed. (#1766) (by @joelim-work)
- **(Breaking)** Line numbers from the `number` and `relativenumber` options are displayed in the main window only, instead of all windows. (#1789) (by @joelim-work)
- **(New)** Support for UNIX domain sockets (for communicating with the `lf` server) is added for Windows. (#1637) (by @Catalyn45)
- **(New)** Color and icon configurations now support the `target` keyword for symbolic links. (#1644) (by @lorentzforces)
- **(New)** A new option `roundbox` is added to use rounded corners when `drawbox` is enabled. (#1653) (by @J-Kappes)
- **(New)** A new option `watch` is added to allow using filesystem notifications to detect and display changes to files. This is an alternative to the `period` option, which polls the filesystem periodically for changes. (#1667) (by @joelim-work)
- **(New)** Icons can now be colored independently of the filename. (#1674) (by @wodesuck)
- **(New)** The `info` option now supports `perm`, `user` and `group` to display the permissions, user and group respectively for each file. (#1799) (by @owallb)
- **(New)** A new option `showbinds` is added to toggle whether the keybinding hints are shown when a keybinding is partially typed. (#1815) (by @dazzywi)
- **(Fix)** Sorting by extension is fixed for hidden files. (#1670) (by @Limero)
- **(Fix)** The `on-quit` hook command is now triggered when the terminal is closed. (#1681) (by @joelim-work)
- **(Fix)** Previews no longer flicker when deleting files. (#1691) (by @joelim-work)
- **(Fix)** Previews no longer flicker when directories are being reloaded. (#1697) (by @joelim-work)
- **(Fix)** `lfcd.nu` now runs properly without raising errors. (#1728) (by @unsafe-andrew)
- **(Fix)** Image previews (composed of ASCII art) containing long lines should now display properly. (#1737) (by @joelim-work)
- **(Fix)** The performance is improved when copying files. (#1749) (by @MahouShoujoMivutilde)
- **(Fix)** `lfcd.cmd` now handles directories with special characters. (#1772) (by @Jaehaks)
- **(Fix)** Icon colors are no longer clipped when displaying in Windows Terminal. (#1777) (by @joelim-work)
- **(Fix)** The file stat info is now cleared when changing to an empty directory. (#1808) (by @joelim-work)
- **(Fix)** Error messages are cleared when opening files. (#1809) (by @joelim-work)
```

---

In the future, I hope that releases can be made more often - six months is quite a long time to wait for bug fixes and other improvements. However I am also much more busier these days, and have less time than before to dedicate to maintaining `lf`. However for now I am happy to review pull requests and submit minor patches, although I don't see myself working on any kind of major changes.

**Comment by DusanLesan** [comment]
Nice to see the new release getting ready! I will switch to master branch, add some new configs, and let you know if I encounter any issues between freeze and release.
<details><summary>Testing config diff. Let me know if some other things could be added</summary>
<p>

```
set drawbox true
set roundbox true
set relativenumber true
set showbinds false

# using icons_colored.example
set icons true

# hook command which is invoked when the terminal gains focus
cmd on-focus-gained :{{
	set cursorparentfmt "\033[7m"
	set cursoractivefmt "\033[7m"
	set cursorpreviewfmt "\033[4m"
}}

# hook command which is invoked when the terminal loses focus
cmd on-focus-lost :{{
	set cursorparentfmt "\033[7;2m"
	set cursoractivefmt "\033[7;2m"
	set cursorpreviewfmt "\033[4;2m"
}}

on-focus-gained

cmd on-quit &{{
	notify-send "Lf quit"
}}
```

</p>
</details>

**Reply by joelim-work** [reply]
Your new config looks fine.

For me the most complex and risky change is adding `watch` in #1667 - during development I tried a number of different approaches and encountered many bugs, but now it is working smoothly for me. I have been using it for a few months already, but you are welcome to try it out as well. Anyway `watch` is disabled by default, so users shouldn't encounter any issues relating to this unless they specifically opt-in to it.

**Reply by DusanLesan** [reply]
I've now added `set watch true` as well just in case
